### ISSUE TITLE
Stale/Fresh Leaf mechanism might not actually get used

### ISSUE THREAD

**Issue by eskimor** [paritytech/polkadot#6097]
It is not clear whether a chain reversion actually triggers a `BlockImport` event, as technically no block got imported. We should verify through reading code/docs and trying whether that code is actually needed and if not, get rid of it.

Semantics of `BlockImport` event should in any case get clarified in code and docs on how it behaves on chain reversions.

**Comment by tifecool** [comment]
How can I execute a "chain reversion"?

**Comment by eskimor** [comment]
You need to trigger a dispute that resolves against a candidate. This will trigger a reversion right before the including block. A dispute can be triggered in Zombienet via malus nodes. See [here](https://github.com/paritytech/polkadot/blob/c4f271c694d9fad640083265ff9840360266b175/zombienet_tests/functional/0003-parachains-garbage-candidate.toml) for example.

**Comment by bkchr** [comment]
A re-org is not triggering any block import event. We also don't directly re-org from the point of view of the database. Babe will just start building on a different fork as being told by the `SelectChain` implementation in Polkadot. The Substrate db re-orgs when there is a fork that is better than the current best chain (this is happening this way because of some legacy stuff AFAIK).

**Comment by eskimor** [comment]
Got it, so this answers the question: The code is not actually in use and can be removed.

**Comment by tifecool** [comment]
I'm assuming the status for the `ActivatedLeaves` will be Stale as a chain revertion is still occurring, but the leaves are being encountered again?

**Comment by ordian** [comment]
> I'm assuming the status for the `ActivatedLeaves` will be Stale as a chain revertion is still occurring, but the leaves are being encountered again?

No, `LeafStatus::Stale` is an unreachable state, so we can remove `LeafStatus` completely. As mentioned above, on chain reversion, there won't be an block import notification issued for the new (old) best block.

**Comment by tifecool** [comment]
I'm not sure I'll be able to assist with this issue 😓
